src/vaadin-time-picker.html

```diff
@@ -377,6 +377,24 @@
           // thus querySelector in textField.focusElement raises an undefined exception on validate
           this.__dropdownElement.addEventListener('value-changed', e => this.__onInputChange(e));
           this.__inputElement.addEventListener('keydown', this.__onKeyDown.bind(this));
+
+          this.__dropdownElement.addEventListener('change', e => {
+            if (e.composedPath()[0] === this.__inputElement) {
+              if (e.detail.sourceEvent.__fromClearButton) {
+                // Clear button clicked. Delay change event until actual value change
+                this.__pendingChange = true;
```

Comment:
Hmm...I can comment out this row but still the whole test suite passes (even "should fire change on clear button click").

Also the same thing with the lines below (with `this.__pendingChange = false;`).

Reply:
Good catch. Updated and simplified logic to listen for 'change' event from `vaadin-combo-box-light`.
Looks like it forwards event in all the cases so we only need to filter out duplicate events from text-field.